extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/BeanContainerImpl.java

```diff
@@ -55,23 +57,27 @@ public ManagedContext requestContext() {
 
     static final class DefaultInstanceFactory<T> implements BeanContainer.Factory<T> {
 
-        final Class<T> type;
+        final Constructor<T> typeCtor;
 
         DefaultInstanceFactory(Class<T> type) {
-            this.type = type;
+            try {
+                this.typeCtor = type.getDeclaredConstructor();
```

Comment:
This seems to be the cause of the failures. If you created an instance factory for an invalid type it currently won't fail until you try and use it, now it fails eagerly.

Reply:
Right. I suspected this - as I remember changing the semantics. I was hoping it would be an improvement :) but we can do that another time.

Sorry didn't jump back on this PR as it's really not urgent, I'll get back to this one when I'll be too tired to do something more useful.